components/listEntry.js

```diff
@@ -1,32 +1,94 @@
-import React from "react";
+import { useState } from "react";
 import styled from "styled-components";
 
 export default function ListEntry({
-  id,
-  time,
-  location,
-  offender,
-  incident,
+  data,
+  incidents,
+  setIncidents,
   handleRemove,
 }) {
+  const [isEditable, setIsEditable] = useState(false);
+  const [location, setLocation] = useState(data.location);
+  const [time, setTime] = useState(data.time);
+  const [offender, setOffender] = useState(data.offender);
+  const [incident, setIncident] = useState(data.incident);
+  function handleSave() {
+    setIncidents(
+      incidents.map((e) =>
+        e.id === data.id
+          ? { id: data.id, location, time, offender, incident }
+          : e
+      )
+    );
+    setIsEditable(false);
+  }
```

Comment:
Is this whole function not an `update` rather than just a `save` ? just asking.

Reply:
It saves the update.😅